pages/_app.js

```diff
@@ -1,12 +1,47 @@
 import GlobalStyle from "../styles";
-
+import { useState } from "react";
 import { books } from "../lib/books.js";
 
 export default function App({ Component, pageProps }) {
+  const [cartItems, setCartItems] = useState([]);
+
+  function addToCart(bookId) {
+    const existigCartItem = cartItems.find(
+      (cartItem) => cartItem.bookId === bookId
+    );
+    if (existigCartItem) {
+      if (existigCartItem.quantity < 5) {
```

Comment:
Just to understand it better: Why exactly is this condition necessary? (current quantity is less than 5..)

Reply:
Because i added the condition of not buying more than 5 books. It could be < 100 e.g. 🙂

Reply:
Oh, that means I can add more than 5 books to the shopping cart, but I can't pay for more than 5. Have I understood that correctly?

Reply:
No, it means you cannot add more than 5 copies of the book to your shopping cart.

Reply:
Ah ok it's about the copies of a single book. Now it all makes sense, thank you!

Reply:
😊